packages/plugin-hardhat/src/index.ts

```diff
@@ -116,3 +116,15 @@ extendEnvironment(hre => {
     };
   });
 });
+
+extendConfig((config: HardhatConfig) => {
+  for (const compiler of config.solidity.compilers) {
+    if (compiler.settings.outputSelection['*']['*'] === undefined) {
+      compiler.settings.outputSelection['*']['*'] = [];
+    }
```

Comment:
`??=` is a very useful operator, it does exactly the same.

```suggestion
    compiler.settings.outputSelection['*']['*'] ??= [];
```